### PR TITLE
Add factory to deploy auctions

### PR DESCRIPTION
**Pull request by PiVortex**

### CODE REVIEW

**Comment by PiVortex** [comment]
@race-of-slothes are you here?

File: factory/tests/tests_basics.rs
Comment (garikbesson) [inline factory/tests/tests_basics.rs:141]: I love the idea of moving such big function chains in separate methods.

I will apply the same approach to my code

File: factory/Cargo.toml
Comment (garikbesson) [inline factory/Cargo.toml:20]: The latest `near-workspaces` version is `0.14.0`. Could you try to update it in dependencies?

File: factory/Cargo.toml
Comment (garikbesson) [inline factory/Cargo.toml:16]: The latest `near-sdk` version is 5.5.0. Could you try to update it in dependencies?

**Review by garikbesson — CHANGES_REQUESTED** [review]

**Comment by garikbesson** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@PiVortex Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!


[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/PiVortex?type=bot&pr=near-examples/auctions-tutorial/14&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/PiVortex?type=bot&pr=near-examples/auctions-tutorial/14&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/PiVortex)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/PiVortex)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: factory/tests/tests_basics.rs
Comment (PiVortex) [reply factory/tests/tests_basics.rs:1]: Update dependencies might have resolved this, if not update your cargo

**Comment by PiVortex** [comment]
Changes made @garikbesson note that tests are failing, these errors are resolved in #15

**Review by garikbesson — APPROVED** [review]

**Comment by race-of-sloths** [comment]
🥁  __Score it!__

@garikbesson, please score the PR with `@race-of-sloths score [1/2/3/5/8/13]`. The contributor deserves it.
If no scoring is provided within 24 hours, this PR will be scored as 2 🦥

**Pull request merged**
